deepchem/models/torch_models/layers.py

```diff
@@ -1286,3 +1286,241 @@ def forward(
+  """
+
+  def __init__(self,
+               atom_types: Union[ArrayLike, torch.Tensor] = None,
+               radial_params: Union[ArrayLike, torch.Tensor] = list(),
+               box_size: Union[ArrayLike, torch.Tensor] = None,
+               **kwargs):
+    """Initialize this layer.
+
+    Parameters
+    ----------
+    atom_types : Union[ArrayLike, torch.Tensor], optional
+      List of atom types.
+    radial_params : Union[ArrayLike, torch.Tensor], optional
+      List of radial params.
+    box_size : Union[ArrayLike, torch.Tensor], optional
+      Length must be equal to the number of features.
+    """
+
+    super(AtomicConvolution, self).__init__(**kwargs)
+    self.atom_types = atom_types
+    self.radial_params = radial_params
+    self.box_size = torch.tensor(box_size) if box_size is not None else None
+
+    vars = []
+    for i in range(3):
+      val = np.array([p[i] for p in self.radial_params]).reshape((-1, 1, 1, 1))
+      vars.append(torch.tensor(val, dtype=torch.float))
+    self.rc = nn.Parameter(vars[0])
+    self.rs = nn.Parameter(vars[1])
+    self.re = nn.Parameter(vars[2])
+
+  def __repr__(self):
+    return (
+        f'{self.__class__.__name__}(atom_types={self.atom_types}, radial_params={self.radial_params}, box_size={self.box_size}, rc={self.rc}, rs={self.rs}, re={self.re})'
+    )
+
+  def forward(self, inputs: Sequence[Union[ArrayLike,
+                                           torch.Tensor]]) -> torch.Tensor:
+    """Invoke this layer.
+
+    B, N, M, d = batch_size, max_num_atoms, max_num_neighbors, num_features
+
+    Parameters
+    ----------
+    inputs: Sequence[Union[ArrayLike, torch.Tensor]]
+      First input are the coordinates/features, of shape (B, N, d)
+      Second input is the neighbor list, of shape (B, N, M)
+      Third input are the atomic numbers of neighbor atoms, of shape (B, N, M)
+
+    Returns
+    -------
+    torch.Tensor of shape (B, N, l)
+      Output of atomic convolution layer.
+
+    Raises
+    ------
+    ValueError
+      When the length of `inputs` is not equal to 3.
+    """
+    if len(inputs) != 3:
+      raise ValueError(f"`inputs` has to be of length 3, got: {len(inputs)}")
+
+    X = torch.tensor(inputs[0])
+    Nbrs = torch.tensor(inputs[1], dtype=torch.int64)
+    Nbrs_Z = torch.tensor(inputs[2])
+
+    B, N, d = X.shape
+    M = Nbrs.shape[-1]
+
+    D = self.distance_tensor(X, Nbrs, self.box_size, B, N, M, d)
+    R = self.distance_matrix(D)
+    R = torch.unsqueeze(R, 0)
+    rsf = self.radial_symmetry_function(R, self.rc, self.rs, self.re)
+
+    if not self.atom_types:
+      cond = torch.not_equal(Nbrs_Z, 0).to(torch.float).reshape((1, -1, N, M))
+      layer = torch.sum(cond * rsf, 3)
+    else:
+      sym = []
+      for i in range(len(self.atom_types)):
+        cond = torch.eq(Nbrs_Z, self.atom_types[i]).to(torch.float).reshape(
+            (1, -1, N, M))
+        sym.append(torch.sum(cond * rsf, 3))
+      layer = torch.concat(sym, 0)
```

Comment:
Do you mind explaining this section? I don't follow what `sym` is doing here

Reply:
To my understanding, those are the symmetries for each atom type in `atom_types` instead of all non-zero elements (which happens in the `if`).

I am not too sure though, which is why I wanted to make a comment on this PR, asking if someone more familiar with the original could confirm or deny that.
I have been having network issues and it went down while making the PR, I am surprised the PR even went up so I didn't get to make that comment.